libs/server/ACL/AccessControlList.cs

```diff
@@ -18,16 +18,31 @@ public class AccessControlList
         /// </summary>
         const string DefaultUserName = "default";
 
+        /// <summary>
+        /// Arbitrary Key for new user lock object.
+        /// </summary>
+        const string NewUserLockObjectKey = "441a61e2-4d4e-498e-8ca0-715cf550e5be";
+
         /// <summary>
         /// Dictionary containing all users defined in the ACL
         /// </summary>
         ConcurrentDictionary<string, User> _users = new();
 
+        /// <summary>
+        /// Dictionary containing stable lock objects for each user in the ACL.
+        /// </summary>
+        ConcurrentDictionary<string, object> _userLockObjects = new();
+
         /// <summary>
         /// The currently configured default user (for fast default lookups)
         /// </summary>
         User _defaultUser;
 
+        /// <summary>
+        /// The <see cref="RespServerSession"/>s that will receive access control list change notifications.
+        /// </summary>
+        private readonly ConcurrentDictionary<string, RespServerSession> _subscribedSessions = new();
```

Comment:
nit: it's not clear to me why this is keyed by a string?  You could just use the `RespServerSesssion` directly, object identity should be sufficient.